### ISSUE TITLE
`nix-build` creates wrong executable

### ISSUE THREAD

**Issue by quinn-dougherty** [plclub/hs-to-coq#192]
Hi! Really psyched about getting up and running with this

I ran `nix-build -A haskellPackages.hs-to-coq` and codegen'd `hs-to-coq/result/bin/hs-to-coq -e hs-to-coq/base/edits --iface-dir hs-to-coq/base -e edits src/Queue.hs -o src-coq`, just as it says in [queue tutorial](https://www.cis.upenn.edu/~plclub/blog/2020-10-09-hs-to-coq/). 

The executable created by `nix-build` happened to codegen `Require GHC.Maybe.` and `GHC.Maybe.Maybe` instead of `option` (in `src/Queue.v`). When I eventually got `stack` to work (from within `nix-shell -A coqPackages.hs-to-coq`, btw) and redid the codegen, I saw you you were using `option`. I knew there was a problem because there's no `Maybe.v` in `GHC` subdir, but the codegen seemed to think there would be! 

I'm a bit of a nix noob myself otherwise I'd fix it myself.

**Comment by nomeata** [comment]
Hmm, I think I encountered the same thing when I recently did something with `hs-to-coq`, and it may be a subtle difference in how GHC works under stack or nix: The file `base/edits` contains edits that rewrite `Maybe` to `option`, but they somehow didn’t work for me, and I worked around it like this: https://github.com/plclub/hs-to-coq/pull/188/files#diff-c9afbeb0e9dd2eb1b6250d41cf4aad9b3d21f4864c0bd1bc2607004399710a5aR17

If you are are nood I suggest to stick to the official way of buliding it (with `stack`). At least until building with nix is “officially supported” (i.e. tested on CI, and the necessary code in `common.mk` so that within a nix-shell, stack is not used).
